template/cordova/lib/MSBuildTools.js

```diff
@@ -171,6 +182,54 @@ function getAvailableUAPVersions() {
     });
 
     return result;
+};
+
+/**
+ * Lists all VS 2017+ instances dirs in ProgramData
+ * 
+ * @return {String[]} List of paths to all VS2017+ instances
+ */
+function getWillowProgDataPaths() {
+    if (!process.env.systemdrive) {
+        // running on linux/osx?
+        return [];
+    }
+    var instancesRoot = path.join(process.env.systemdrive, 'ProgramData/Microsoft/VisualStudio/Packages/_Instances');
+    if (!shell.test('-d', instancesRoot)) {
+        // can't seem to find VS instances dir, return empty result
+        return [];
+    }
+
+    return fs.readdirSync(instancesRoot).map(function(file) {
```

Comment:
`filter` would leave it as a relative path, but we need to make an absolute one.